collectors/python.d.plugin/anomalies/anomalies.chart.py

```diff
@@ -0,0 +1,312 @@
+                    for chart in self.custom_models_host_charts_dict[host]:
+                        if chart not in self.host_charts_dict[host]:
+                            self.host_charts_dict[host].extend(chart)
+        else:
+            self.models_in_scope = [f'{self.host}::{c}' for c in self.charts_in_scope]
+            self.host_charts_dict = {self.host: self.charts_in_scope}
+        self.model_display_names = {model: model.split('::')[1] if '::' in model else model for model in self.models_in_scope}
+
+        self.train_max_n = self.configuration.get('train_max_n', 100000)
+        self.train_n_secs = self.configuration.get('train_n_secs', 14400)
+        self.offset_n_secs = self.configuration.get('offset_n_secs', 0)
+        self.train_every_n = self.configuration.get('train_every_n', 900)
+        self.initial_train_data_after = self.configuration.get('initial_train_data_after', 0)
+        self.initial_train_data_before = self.configuration.get('initial_train_data_before', 0)
+        self.contamination = self.configuration.get('contamination', 0.001)
+        self.lags_n = {model: self.configuration.get('lags_n', 5) for model in self.models_in_scope}
+        self.smooth_n = {model: self.configuration.get('smooth_n', 5) for model in self.models_in_scope}
+        self.diffs_n = {model: self.configuration.get('diffs_n', 5) for model in self.models_in_scope}
+
+        self.model = self.configuration.get('model', 'pca')
+        if self.model == 'pca':
+            self.models = {model: PCA(contamination=self.contamination) for model in self.models_in_scope}
+        elif self.model == 'loda':
+            self.models = {model: LODA(contamination=self.contamination) for model in self.models_in_scope}
+        elif self.model == 'iforest':
+            self.models = {model: IForest(n_estimators=50, bootstrap=True, behaviour='new', contamination=self.contamination) for model in self.models_in_scope}
+        elif self.model == 'cblof':
+            self.models = {model: CBLOF(n_clusters=3, contamination=self.contamination) for model in self.models_in_scope}
+        elif self.model == 'feature_bagging':
+            self.models = {model: FeatureBagging(base_estimator=PCA(contamination=self.contamination), contamination=self.contamination) for model in self.models_in_scope}
+        elif self.model == 'copod':
+            self.models = {model: COPOD(contamination=self.contamination) for model in self.models_in_scope}
+        elif self.model == 'hbos':
+            self.models = {model: HBOS(contamination=self.contamination) for model in self.models_in_scope}
+        else:
+            self.models = {model: HBOS(contamination=self.contamination) for model in self.models_in_scope}
+        self.custom_model_scalers = {model: MinMaxScaler() for model in self.models_in_scope}
+
+        self.fitted_at = {}
+        self.df_allmetrics = pd.DataFrame()
+        self.data_latest = {}
+        self.expected_cols = []
```

Comment:
How about to split init into `init_this`, `init_that`? So it is easier to follow what is happening there.

Reply:
Not sure what you mean here - separate init type functions for each block of related stuff or ??

Reply:
yeah, i noticed that `__init__` is big and hard to read, just assumed that it would be nice to move some logic into a separate function. Does it make sense now?

Reply:
makes sense - will have a go at this.